soc/xtensa/intel_adsp/ace/multiprocessing.c

```diff
@@ -82,9 +84,21 @@ void soc_start_core(int cpu_num)
 	int retry = CORE_POWER_CHECK_NUM;
 
 	if (cpu_num > 0) {
+		extern void dsp_restore_vector(void);
 		/* Initialize the ROM jump address */
 		uint32_t *rom_jump_vector = (uint32_t *) ROM_JUMP_ADDR;
-		*rom_jump_vector = (uint32_t) z_soc_mp_asm_entry;
+
+		/* We need to find out what type of booting is taking place here. Secondary cores
+		 * can be disabled and enabled multiple times during runtime. During kernel
+		 * initialization, the next pm state is set to ACTIVE. This way we can determine
+		 * whether the core is being turned on again or for the first time.
+		 */
+		if (pm_state_next_get(cpu_num)->state == PM_STATE_ACTIVE) {
```

Comment:
PM is optional, one can build without it. If you build wihout PM, `pm_state_next_get()` will return a NULL pointer. This change can cause a null pointer dereference. Please account this possibility here.